code/_expts/slotnick_etal_2001/de_StimCreate.m

```diff
@@ -0,0 +1,346 @@
+function [train,test] = de_StimCreate(stimSet, taskType, opt)
+%Input:
+%  stimSet  : a string specifying which INPUT sets to train autoencoder on
+%                de: (default) 10 "far", 5 "near", 5 "on" images
+%
+%  taskType : a string specifying which OUTPUT task to train on
+%               categorical = compare on (1) to off (0)
+%               coordinate = compare near (1) vs. far (0)
+%
+%  opt      : a vector of options; all listed will be applied
+%
+%OUTPUT: a data file with the following variables:
+%
+%  train.X    : matrix containing 20 vectors, each a unique hierarchical stimulus.
+%  train.T    : target vectors for perceptron (labels, based on task)
+%
+%  test.*     : same as train object, but
+
+  if (~exist('opt','var')),      opt      = {};     end;
+  if (~iscell(opt)),             opt      = {opt};  end;
+  
+  if (~exist('stimSet', 'var')), stimSet  = 'de';   end;
+
+  % Create the input images.
+  train.nInput = [68 50];
+  train.X = zeros(prod(train.nInput), 0);
+  train.XLAB = cell(20, 1);
+  train.TLAB = cell(20, 1);
+  %train.XLAB = guru_csprintf('height=%.2f', num2cell(heights);
+  counter = 0;
+  heights = [];
+  if (strcmp(stimSet, 'de')) %default set of images
+      figure('Position', [0, 0, 1024, 800]);
+      % first, 10 "far" images
+      for ii=1:5
+          distances = [10, 12];
+          for ij=1:length(distances)
+              img = blob_stimuli(distances(ij), 3, ii);
+              train.X(:, end+1) = reshape(img, prod(train.nInput), 1);
+              counter = counter + 1;
+              train.XLAB{counter} = sprintf('%dpx from %d%c', distances(ij), (ii-1)* 72, char(176));
+              subplot(4, 5, counter);
+              imshow(img);
+              title(train.XLAB{counter});
+              heights = [heights, distances(ij)];
+          end
+      end
+      % now, 10 "close" images (5 on, 5 off)
+      
+      for ii=1:5
+          distances = [0, 4];
+          for ij=1:length(distances)
+              counter = counter + 1;
+              img = blob_stimuli(distances(ij), 3, ii);
+              train.XLAB{counter} = sprintf('%dpx from %d%c', distances(ij), (ii-1)* 72, char(176));
+              subplot(4, 5, counter);
+              imshow(img);
+              title(train.XLAB{counter});
+              train.X(:, end+1) = reshape(img, prod(train.nInput), 1);
+              heights = [heights, distances(ij)];
+          end
+          
+      end
+      test = train;
+  else
```

Comment:
For readability, I prefer to put short blocks from an `if...else` first. I suggest changing the `if` check above to check `if ~strcmp... then error.`. Doing that, you also don't need to indent for the `else` statement, since if the `if` statement is entered, you won't continue the function (it errors).